### DISCUSSION TITLE
Recommended distro for MooseFS?

### DISCUSSION THREAD

**Discussion by Lindsay-Mathieson** [moosefs/moosefs#595] [Q&A]
Looking at starting a test environment for vs 4, would be done via LXC on proxmox

Is there a preferred distro? my normal preference is apt based, specifically Debian.

**Comment by xandrus** [comment]
Hi Lindsay!

You can easily start testing MooseFS with LXC on Proxmox, but you must remember to enable FUSE support in the container features.

When creating an LXC container, uncheck the “Unprivileged container” checkbox.
<img src="https://github.com/user-attachments/assets/1733bd3d-39ba-4c48-a514-fa0a4308c321" width="400">

After completing container creation go to: `GUI > Container > Options > Features` and select FUSE.
<img src="https://github.com/user-attachments/assets/8cce6579-cb7a-4c7f-895d-772b0ff41bc9" width="400">

**I would also like to add that the FUSE option is only required for the container with the MooseFS client**

Regarding Linux distribution. There is no particular distribution that is preferred over others.
All modern Linux operating systems will work really well. If you prefer deb packaging system, you can easily use Debian12 or Ubuntu 24.04

For example repository for Debian 12, AMD64 can be configured as follows:
```
curl https://repository.moosefs.com/moosefs.key | gpg -o /etc/apt/keyrings/moosefs.gpg --dearmor
echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/moosefs.gpg] http://repository.moosefs.com/moosefs-4/apt/debian/bookworm bookworm main" > /etc/apt/sources.list.d/moosefs.list
apt update
```
And then easily install MooseFS packages using the command: `apt install moosefs-[component name]` 
By the way, please check the available package installation options on the MooseFS website: https://moosefs.com/download/#current

**Reply by Lindsay-Mathieson** [reply]
Thanks! Looking fwd to checking out the 4.x release

**Reply by Lindsay-Mathieson** [reply]
It all working very well! I ended up migrating my entire ceph setup to moosefs, took a while with a lot disk juggling :) but I now have 8 disks spread between 6 chunkservers for a total of 55TB of storage. RAM usage and cpu load is greatly reduced.

Am using ec4+1, which so greatly reduced disk usage as well, has dropped from 43% to 27%, immensly pleased with the result.

Its a media server (Plex), and its happily serving up multiple high bitrate streams.

> You can easily start testing MooseFS with LXC on Proxmox, but you must remember to enable FUSE support in the container features.

I ended up doing the mfsmount on the host (proxmox) and passing it through via bind mount, as lxc's can't backed up when they have a fuse mount.

> When creating an LXC container, uncheck the “Unprivileged container” checkbox.

Any particular reason for priviliged? I'm using unpriviliged lxc's with the diskmounts passed through as bind mounts. Seems to be working fine.

**Reply by xandrus** [reply]
As I mentioned earlier - privileged LXC containers with FUSE option are required only for containers with MooseFS client. Because mfsmount depends on the fuse kernel module.

**Reply by Lindsay-Mathieson** [reply]
> As I mentioned earlier - privileged LXC containers with FUSE option are required only for containers with MooseFS client. Because mfsmount depends on the fuse kernel module.

With Proxmox you can definitely use fuse with an unprivileged container, you just have to check the "Fuse" feature in the lxc options.

The drawback, for ZFS backed containers is that snapshot backups will hang. Thats why I mounted mfs on the proxmox host and passed it through via a bind mount.

**Reply by xandrus** [reply]
@LindsayMathieson you are absolutely right!
The FUSE function can be used on an unprivileged container without any problems!
My misunderstanding!
